corehq/apps/enterprise/iterators.py

```diff
@@ -0,0 +1,9 @@
+def raise_after_max_elements(it, max_elements, exception=None):
+    total_yielded = 0
+    for ele in it:
+        if total_yielded >= max_elements:
+            exception = exception or Exception('Too Many Elements')
+            raise exception
+
+        yield ele
+        total_yielded += 1
```

Comment:
FYI `enumerate` is lazy, so this won't force up-front evaluation of the iterable:
```suggestion
    for total_yielded, ele in enumerate(it)
        if total_yielded >= max_elements:
            exception = exception or Exception('Too Many Elements')
            raise exception

        yield ele
```

Reply:
Adopted this in efc087282fc87ecdb559e0a1695fc41f3e44e6e1